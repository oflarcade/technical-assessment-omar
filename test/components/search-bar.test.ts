import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import SearchBar from '~/components/rick-and-morty/search-bar.vue';

describe('SearchBar.vue', () => {
  it('renders correctly with the given searchTerm', () => {
    const wrapper = mount(SearchBar, {
      props: {
        searchTerm: 'test',
      },
    });

    const input = wrapper.find('input');
    expect((input.element as HTMLInputElement).value).toBe('test');
  });

  it('emits update:searchTerm event with the correct value on input', async () => {
    const wrapper = mount(SearchBar, {
      props: {
        searchTerm: '',
      },
    });

    const input = wrapper.find('input');
    await input.setValue('Rick');

    expect(wrapper.emitted('update:searchTerm')).toHaveLength(1);
    expect(wrapper.emitted('update:searchTerm')![0]).toEqual(['Rick']);
  });

  it('matches snapshot', () => {
    const wrapper = mount(SearchBar, {
      props: {
        searchTerm: 'test',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
