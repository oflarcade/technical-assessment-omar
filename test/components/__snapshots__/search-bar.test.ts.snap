// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar.vue > matches snapshot 1`] = `"<div class="relative"><input type="text" placeholder="Search characters..." class="w-full px-4 py-2 rounded-lg bg-gray-100 focus:outline-none focus:bg-white text-black border border-gray-300"></div>"`;
